### PR TITLE
No more Oversized Piggyback riding v2.0

### PR DESCRIPTION
**Pull request by Microvolnovka19**
## About The Pull Request

I've added cool totally not stolen from a vending machine code squishing mechanic:

https://user-images.githubusercontent.com/59139863/192481324-bb8d69f5-610f-4ef4-baa5-d5883fc07a76.mp4

## How This Contributes To The Skyrat Roleplay Experience

Fun vendingmachine-like squish lets you feel the full weight of your oversized partner

## Changelog

:cl: SuperDrish
add: Oversized Piggyback riding will now squish(like the vending machine does) the smaller creature
/:cl:

### CODE REVIEW

File: modular_skyrat/modules/customization/modules/mob/living/carbon/human/human_defines.dm [hunk 1]
```diff
@@ -17,5 +17,6 @@
 	var/oversized_piggywound_chance = 50
 	/// Base damage for oversized piggyback riding.
 	var/oversized_piggydam = 25
-	/// Knockdown time for oversized piggyback riding in deciseconds. (10 deciseconds = 1 second)
-	var/oversized_piggyknock = 30
+	/// Paralyze time for oversized piggyback riding in deciseconds. (10 deciseconds = 1 second)
+	var/oversized_piggyparalyze = 30
```
Comment (Zonespace27) [inline modular_skyrat/modules/customization/modules/mob/living/carbon/human/human_defines.dm]: use SECONDS defines

Comment (nikita-chekistov) [reply modular_skyrat/modules/customization/modules/mob/living/carbon/human/human_defines.dm]: > use SECONDS defines

So, should I use seconds in defines and just multiply it by ten in the code?

Comment (Tastyfish) [reply modular_skyrat/modules/customization/modules/mob/living/carbon/human/human_defines.dm]: No, there is a `SECONDS` define for times. Use it like `= 3 SECONDS` and the result will be 30.

**Comment by gamermomentum** [comment]
Just to be clear. 
This is gonna add an insane upside to this that is, "Anyone who aggressive grabs you in combat, you can subject to a vending  machine-esque attack, and hardstun them for aaages."?
Piggybacking to get out of aggro grabs is already pretty good but this is wild.

**Comment by tf-4** [comment]
see above, piggybacking to get out of aggro grabs is already extremely good and annoying as shit. This is a pretty substantial buff

**Comment by SpaceVampire** [comment]
I'm certain that our Veteran Crew can be trusted to use this responsibly.
Without any worries about powergaming or it being LRP.
They are the very best of the best.

**Comment by nikita-chekistov** [comment]
> Just to be clear. This is gonna add an insane upside to this that is, "Anyone who aggressive grabs you in combat, you can subject to a vending machine-esque attack, and hardstun them for aaages."? Piggybacking to get out of aggro grabs is already pretty good but this is wild.

Its already added in #15708. This is mostly a decorative update with squish and sound.

**Comment by nikita-chekistov** [comment]
> see above, piggybacking to get out of aggro grabs is already extremely good and annoying as shit. This is a pretty substantial buff

Yes it is, but:
1. You require a Veteran status to use this buff.
2. A grabber still have time to react. It's not instant pelvis breaking techinque

**Comment by tf-4** [comment]
> > Just to be clear. This is gonna add an insane upside to this that is, "Anyone who aggressive grabs you in combat, you can subject to a vending machine-esque attack, and hardstun them for aaages."? Piggybacking to get out of aggro grabs is already pretty good but this is wild.
> 
> Its already added in #15708. This is mostly a decorative update with squish and sound.

... it adds a 3 second paralyse, how is that decorative?

**Comment by nikita-chekistov** [comment]
> > > Just to be clear. This is gonna add an insane upside to this that is, "Anyone who aggressive grabs you in combat, you can subject to a vending machine-esque attack, and hardstun them for aaages."? Piggybacking to get out of aggro grabs is already pretty good but this is wild.
> > 
> > Its already added in #15708. This is mostly a decorative update with squish and sound.
> 
> ... it adds a 3 second paralyse, how is that decorative?

It replaces knockdown with paralyze. But  it could be easily changed back, if that's a problem

**Comment by Zergspower** [comment]
I dont know, given this affects veterans directly I suggest ensuring the vet team can sign off on this. We've had issues of some PR's being bait to abuse

**Comment by Zonespace27** [comment]
[modular_skyrat/modules/oversized/readme.md](https://github.com/Skyrat-SS13/Skyrat-tg/blob/master/modular_skyrat/modules/oversized/readme.md)

Talk to a maintainer if you want this re-opened

File: code/modules/mob/living/carbon/human/human.dm [hunk 2]
```diff
@@ -1016,8 +1016,13 @@
 			to_chat(target, span_danger("You accidentally crush [src]!"))
 		else
 			to_chat(src, span_danger("You hurt your [affecting.name] while trying to endure the weight of [target]!"))
-		apply_damage(oversized_piggydam, BRUTE, affecting, wound_bonus=wound_bon) //Try to lift a 2 centner creature. 
-		Knockdown(oversized_piggyknock)
+		apply_damage(oversized_piggydam, BRUTE, affecting, wound_bonus=wound_bon)
+		playsound(src, 'sound/effects/splat.ogg', 50, TRUE)
+		src.AddElement(/datum/element/squish, 20 SECONDS) // Totally not stolen from a vending machine code
+		src.Knockdown(oversized_piggyknock)
```
Comment (Zonespace27) [inline code/modules/mob/living/carbon/human/human.dm]: unneeded `src`

File: code/modules/mob/living/carbon/human/human.dm [hunk 3]
```diff
@@ -1016,8 +1016,13 @@
 			to_chat(target, span_danger("You accidentally crush [src]!"))
 		else
 			to_chat(src, span_danger("You hurt your [affecting.name] while trying to endure the weight of [target]!"))
-		apply_damage(oversized_piggydam, BRUTE, affecting, wound_bonus=wound_bon) //Try to lift a 2 centner creature. 
-		Knockdown(oversized_piggyknock)
+		apply_damage(oversized_piggydam, BRUTE, affecting, wound_bonus=wound_bon)
+		playsound(src, 'sound/effects/splat.ogg', 50, TRUE)
+		src.AddElement(/datum/element/squish, 20 SECONDS) // Totally not stolen from a vending machine code
```
Comment (Zonespace27) [inline code/modules/mob/living/carbon/human/human.dm]: ditto

**Pull request merged**